pkg/auth/oauth2authenticator.go

```diff
@@ -0,0 +1,252 @@
+package auth
+
+import (
+	"context"
+	crypto_rand "crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"html"
+	"math/rand"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/pkg/browser"
+	"github.com/snyk/go-application-framework/pkg/configuration"
+	"golang.org/x/oauth2"
+)
+
+const (
+	CONFIG_KEY_OAUTH_TOKEN string = "OAUTH_TOKEN"
+	OAUTH_CLIENT_ID        string = "0oa37b7oa3zOoDWCe4h7"
+	CALLBACK_HOSTNAME      string = "127.0.0.1"
+	CALLBACK_PATH          string = "/authorization-code/callback"
+)
+
+var accepted_callback_ports = []int{8080, 18081, 28082, 38083, 48084}
+
+type oAuth2Authenticator struct {
+	httpClient         *http.Client
+	config             configuration.Configuration
+	oauthConfig        *oauth2.Config
+	token              *oauth2.Token
+	headless           bool
+	openBrowserFunc    func(authUrl string)
+	shutdownServerFunc func(server *http.Server)
+}
+
+func init() {
+	var seed int64
+	var b [8]byte
+	_, err := crypto_rand.Read(b[:])
+	if err != nil {
+		seed = time.Now().UnixNano() // fallback to time only if necessary
+	} else {
+		seed = int64(binary.LittleEndian.Uint64(b[:])) // based on https://stackoverflow.com/a/54491783
+	}
+	rand.Seed(seed)
+}
+
+func openBrowser(authUrl string) {
+	_ = browser.OpenURL(authUrl)
+}
+
+func shutdownServer(server *http.Server) {
+	time.Sleep(500)
+	_ = server.Shutdown(context.Background())
+}
+
+func getRedirectUri(port int) string {
+	callback := fmt.Sprintf("http://%s:%d%s", CALLBACK_HOSTNAME, port, CALLBACK_PATH)
+	return callback
+}
+
+func getOAuthConfiguration(config configuration.Configuration) *oauth2.Config {
+
+	appUrl := config.GetString(configuration.WEB_APP_URL)
+	// tokenUrl := strings.Replace(appUrl, "app.", "id.", 1) + "/oauth2/default/v1/token"
```

Comment:
Can this comment be removed? source control will get it back for us, if needed.

Reply:
Then there should be comment with todo/fixme explaining that this is the code to be used in the future :)